### DISCUSSION TITLE
Is it possible to use both LA and UBL on a STM32F401RCT6?

### DISCUSSION THREAD

**Discussion by gitfve** [mriscoc/Ender3V2S1#824] [General]
Is it possible to use **both LA and UBL** on a STM32F401RCT6 (Ender-3 S1 Pro)

I have for now only used the fw with UBL ("**Ender3S1-F4-UBL-20230312.bin**").

I see in config you have a LA and a UBL for the F4 but not both.
https://github.com/mriscoc/Special_Configurations
https://github.com/mriscoc/Special_Configurations/releases/tag/LinearAdvance

Is it the 256KB ROM space there are a problem or can i compile with both?

**Comment by TheWebMachine** [comment]
UBL is huge and a *lot* had to be removed to make room for it on the F4 boards, as the 32-bit bootloader takes up more space than other boards.

**Comment by mriscoc** [comment]
The different chips used on the various creality boards have different amounts of flash ROM for firmware storage, this means that different boards will be limited by ROM size to which firmware features can be enabled while still allowing the firmware to fit. The F4 chip has the smallest amount of usable ROM. The maximum firmware size for the F4 is 192k, any compiled firmwares over 192k will not work due to the mentioned size limitations, the precompiled firmwares for the F4 boards have some minor features removed to allow for stuff like UBL and other new marlin features. Removing enough features might leave room to allow a firmware to be compiled with UBL and LA, but that's a matter of preference on which features to remove.
